src/wallet/wallet2.cpp

```diff
@@ -3424,6 +3447,44 @@ bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_
         m_device_name = m_key_device_type == hw::device::device_type::LEDGER ? "Ledger" : "default";
       }
     }
+    
+    if (json.HasMember("original_keys_available"))
+    {
+      GET_FIELD_FROM_JSON_RETURN_ON_ERROR(json, original_keys_available, int, Int, false, false);
+      m_original_keys_available = field_original_keys_available;
+      if (m_original_keys_available)
+      {
+	GET_FIELD_FROM_JSON_RETURN_ON_ERROR(json, original_address, std::string, String, false, std::string());
+	if (!field_original_address_found)
+	{
+	  LOG_ERROR("Field original_address not found in JSON");
+	  return false;
+	}
```

Comment:
Why do you set it to non mandatory just to bail out if not found ?

Reply:
I am not sure what you mean. In any case, the idea is the following: If `original_keys_available` is there, both `original_address` and `original_view_secret_key` must be there too, or: With `original_keys_available` present it's a structural error if the other two are missing.

Reply:
The penultimate parameter (mandatory) is false, but you add a test to do exactly what it'd do had it been true.

Reply:
Ok, now I see it. Indeed pretty dumb. Corrected.